src/TraceEvent/TraceLog.cs

```diff
@@ -642,6 +642,16 @@ private static void RemoveAllButLastEntries<T>(ref GrowableArray<T> growableArra
         /// </summary>
         private unsafe void DispatchClonedEvent(TraceEvent toSend)
         {
+            realTimeSource.RegisterUnhandledEventImpl(te =>
+            {
+                if (toSend.containsSelfDescribingMetadata)
```

Comment:
Would be good to have a comment here describing the flow of self-describing metadata.  Something along the lines of:

Self describing metadata doesn't make it to `realTimeSource` because `TraceLog` overwrites the event's extended data.  To address this, we use the event from the raw source, which contains the metadata, and convert it to a template that can be registered with `realTimeSource`.

Reply:
👍